tests/extras/datasets/tensorflow/test_tensorflow_model_data_set.py

```diff
@@ -0,0 +1,355 @@
+# limitations under the License.
+
+from pathlib import PurePosixPath
+
+import numpy as np
+import pytest
+from fsspec.implementations.http import HTTPFileSystem
+from fsspec.implementations.local import LocalFileSystem
+from gcsfs import GCSFileSystem
+import s3fs
+import tensorflow as tf
+from moto import mock_s3
+from s3fs import S3FileSystem
+
+from kedro.extras.datasets.tensorflow import TensorFlowModelDataset
+from kedro.io import DataSetError
+from kedro.io.core import Version
+
+AWS_MODEL_DIR_NAME = "test_tf_model"
+AWS_BUCKET_NAME = "test_bucket"
+AWS_CREDENTIALS = dict(
+    aws_access_key_id="FAKE_ACCESS_KEY", aws_secret_access_key="FAKE_SECRET_KEY"
+)
+
+
+@pytest.fixture
+def filepath(tmp_path):
+    return str(tmp_path / "test.tf")
+
+
+@pytest.fixture(params=[None])
+def load_args(request):
+    return request.param
+
+
+@pytest.fixture(
+    params=[
+        {
+            "overwrite": True,
+            "include_optimizer": True,
+            "save_format": "tf",
+            "signatures": None,
+            "options": None,
+        }
+    ]
+)
+def save_args(request):
+    return request.param
+
+
+@pytest.fixture
+def dummy_x_train():
+    return np.array([[[1.0], [1.0]], [[0.0], [0.0]]])
+
+
+@pytest.fixture
+def dummy_y_train():
+    return np.array([[[1], [1]], [[1], [1]]])
+
+
+@pytest.fixture
+def dummy_x_test():
+    return np.array([[[0.0], [0.0]], [[1.0], [1.0]]])
+
+
+@pytest.fixture
+def tensorflow_model_data_set(filepath, load_args, save_args, fs_args):
+    return TensorFlowModelDataset(
+        filepath=filepath, load_args=load_args, save_args=save_args, fs_args=fs_args
+    )
+
+
+@pytest.fixture
+def versioned_tensorflow_model_data_set(filepath, load_args, save_args, version):
+    return TensorFlowModelDataset(
+        filepath=filepath, load_args=load_args, save_args=save_args, version=version
+    )
+
+
+@pytest.fixture
+def dummy_tf_base_model(dummy_x_train, dummy_y_train):
+    # dummy 1 layer model as used in TF tests, see
+    # https://github.com/tensorflow/tensorflow/blob/8de272b3f3b73bea8d947c5f15143a9f1cfcfc6f/tensorflow/python/keras/models_test.py#L342
+    inputs = tf.keras.Input(shape=(2, 1))
+    x = tf.keras.layers.Dense(1)(inputs)
+    outputs = tf.keras.layers.Dense(1)(x)
+
+    model = tf.keras.Model(inputs=inputs, outputs=outputs, name="1_layer_dummy")
+    model.compile("rmsprop", "mse")
+    model.fit(dummy_x_train, dummy_y_train, batch_size=64, epochs=1)
+    # from https://www.tensorflow.org/guide/keras/save_and_serialize
+    # Reset metrics before saving so that loaded model has same state,
+    # since metric states are not preserved by Model.save_weights
+    model.reset_metrics()
+    return model
+
+
+@pytest.fixture
+def dummy_tf_subclassed_model(dummy_x_train, dummy_y_train):
+    """
+    demonstrate that own class models cannot be saved using HDF5 format but can using TF format
+    :param dummy_x_train:
+    :param dummy_y_train:
+    :return:
+    """
```

Comment:
We use Google docstrings instead, can you switch to those? :) It's a test, so you can just do
```suggestion
    """Demonstrate that own class models cannot be saved using HDF5 format but can using TF format"""
```

or 

```suggestion
    """Demonstrate that own class models cannot be saved
    using HDF5 format but can using TF format
    """
```

if it complains about length of the line. :)